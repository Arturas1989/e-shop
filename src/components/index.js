export { Header } from './Header/Header';
export { NavigationMenu } from './Header/Menu/NavigationMenu/NavigationMenu';
export { LoginMenu } from './Header/Menu/LoginMenu/LoginMenu';
export { Hamburger } from './Header/Hamburger/Hamburger';
export { CloseMenu } from './Header/Menu/NavigationMenu/CloseMenu';
export { LoginLink } from './Header/Menu/LoginMenu/LoginLink';

export { SearchBar } from './SearchBar/SearchBar';
export { InputBar } from './SearchBar/InputBar';

export { Footer } from './Footer/Footer';

export { PrimaryButton } from './Buttons/PrimaryButton/PrimaryButton';
export { SecondaryButton } from './Buttons/SecondaryButton/SecondaryButton';

export { Rating } from './ProductCard/Rating/Rating';
export { BestSeller } from './ProductCard/BestSeller/BestSeller';
export { Heading } from './Heading/Heading';
