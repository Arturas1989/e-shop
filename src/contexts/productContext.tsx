import { ReactNode, createContext, useContext } from "react";
import { DataInfo, Product } from "../types";
import { useData } from "../hooks";

type ProductContextProviderProps = {
  children: ReactNode;
};

type ProductContextValue = {
  products: Product[] | null;
  setProducts: React.Dispatch<React.SetStateAction<Product[] | null>>;
  isProductsLoading: boolean;
  featuredProducts: Product[] | null;
  setFeatured: React.Dispatch<React.SetStateAction<Product[] | null>>;
  isFeaturedProductsLoading: boolean;
} | null;

const ProductContext = createContext<ProductContextValue>(null);

export const ProductContextProvider = ({ children }: ProductContextProviderProps) => {
  const productsInfo: DataInfo = {
    collectionName: 'products',
    onlyFeatured: false,
  }
  const [products, isProductsLoading, setProducts] = useData<Product>(productsInfo);
  const featuredInfo: DataInfo = {
    collectionName: 'products',
    onlyFeatured: true
  }
  const [featuredProducts, isFeaturedProductsLoading, setFeatured] = useData<Product>(featuredInfo);

  const contextVal = {
    products, isProductsLoading, setProducts, featuredProducts, setFeatured, isFeaturedProductsLoading
  }
  return (
    <ProductContext.Provider value={contextVal}>
      {children}
    </ProductContext.Provider>
  )
};

export const useProductsContext = () => useContext(ProductContext);
