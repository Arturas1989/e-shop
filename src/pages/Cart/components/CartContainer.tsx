import { SecondaryButton } from "../../../components";
import { Product } from "../../../types";
import { CartCard } from "./Cartcard/CartCard";
import { Total } from "./Total";

type CartContainerProps = {
  products: Product[] | null;
};

export const CartContainer = ({ products }: CartContainerProps) => {
  return (
    <>
      {products?.map((product) => (
        <CartCard key={product.id} product={product} />
      ))}
      <Total products={products} />
      <SecondaryButton>
        Place order<i className="bi bi-arrow-right"></i>
      </SecondaryButton>
    </>
  );
};
