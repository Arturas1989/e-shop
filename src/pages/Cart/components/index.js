export { ProductsCount } from './ProductsCount';
export { CartCard } from './Cartcard/CartCard';
export { CartProducts } from './CartProducts';
