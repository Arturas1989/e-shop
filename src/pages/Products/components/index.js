export { FilterOpenButton } from './Filter/FilterOpenButton';
export { ProductsCount } from './ProductsCount';
export { Filter } from './Filter/Filter';
export { InputGroup } from './Filter/InputGroup';
