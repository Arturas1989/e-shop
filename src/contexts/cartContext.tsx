import { ReactNode, createContext, useContext } from "react";
import { Cart } from "../services/Cart";
import { useCart } from "../hooks";

type CartContextProviderProps = {
  children: ReactNode;
};

type CartContextValue = {
  cart: Cart | null;
  setCart: React.Dispatch<React.SetStateAction<Cart | null>>;
  isLoading: boolean;
} | null;

const CartContext = createContext<CartContextValue>(null);

export const CartContextProvider = ({ children }: CartContextProviderProps) => {
  const {cart, setCart, isLoading} = useCart();
  return (
    <CartContext.Provider value={{cart, setCart, isLoading}}>
      {children}
    </CartContext.Provider>
  )
};

export const useCartContext = () => useContext(CartContext);
