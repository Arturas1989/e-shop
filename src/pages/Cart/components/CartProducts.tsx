import { Loading, SecondaryButton } from '../../../components';
import { useCartContext } from '../../../contexts/cartContext';
import { useCartProducts } from '../../../hooks';
import { CartCard } from './Cartcard/CartCard';
import { Total } from './Total';

type CartProductsProps = {
  name?: string;
};

export const CartProducts = ({ name }: CartProductsProps) => {
  const { cart } = useCartContext()!;
  const [isLoading, products] = useCartProducts(cart);
  console.log(products);
  return (
    <section className="cards">
      {isLoading ? (
        <Loading />
      ) : (
        <>
          {products?.map((product) => (
            <CartCard key={product.id} product={product} />
          ))}
          <Total products={products} />
          <SecondaryButton>
            Place order<i className="bi bi-arrow-right"></i>
          </SecondaryButton>
        </>
      )}
    </section>
  );
};
