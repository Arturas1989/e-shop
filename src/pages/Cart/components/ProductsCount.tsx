import { useCartContext } from "../../../contexts/cartContext";
import { useCartProducts } from "../../../hooks";

export const ProductsCount = () => {
  const {cart} = useCartContext()!;
  const [, products] = useCartProducts(cart);

  return (
    <span>({products?.length})</span>
  )
};
