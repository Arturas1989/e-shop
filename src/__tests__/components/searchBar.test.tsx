import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Header, SearchBar } from '../../components';
import { CartContextProvider } from '../../contexts/cartContext';
import { ProductContextProvider } from '../../contexts/productContext';
import { renderHeader } from './loginMenu.test';

describe('test description', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <CartContextProvider>
          <ProductContextProvider>
            <BrowserRouter>
              <SearchBar />
            </BrowserRouter>
          </ProductContextProvider>
        </CartContextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('search bar visible class changes correctly in the header by clicking header search icon', () => {
    renderHeader();
    const header = screen.getByTestId('header');
    expect(header).toBeInTheDocument();

    expect(header).toHaveClass('search-not-visible');

    const headerSearchIcon = screen.getByTestId('search-icon');
    expect(headerSearchIcon).toBeInTheDocument();

    fireEvent.click(headerSearchIcon);
    expect(header).not.toHaveClass('search-not-visible');
  });

  test('search bar visible class changes correctly in the header by clicking menu search icon', () => {
    renderHeader();
    
    const header = screen.getByTestId('header');
    expect(header).toBeInTheDocument();

    expect(header).toHaveClass('search-not-visible');

    const menuSearchIcon = screen.getByTestId('search-icon2');
    expect(menuSearchIcon).toBeInTheDocument();

    fireEvent.click(menuSearchIcon);
    expect(header).not.toHaveClass('search-not-visible');
  });
});
