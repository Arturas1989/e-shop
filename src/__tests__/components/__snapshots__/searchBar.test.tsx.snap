// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test description renders correctly 1`] = `
<section
  className="search"
>
  <div
    className="search-bar"
    data-testid="search-bar"
  >
    <i
      className="bi bi-search"
    />
    <input
      className="input-bar"
      placeholder="Search"
      type="text"
    />
    <button
      className="secondary-button"
    >
      <i
        className="bi bi-search"
      />
    </button>
  </div>
</section>
`;
